server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -294,7 +294,13 @@ public void testWriteFullMetadataFailureForGlobalMetadata() throws IOException {
         ArgumentCaptor<ActionListener<Void>> actionListenerArgumentCaptor = ArgumentCaptor.forClass(ActionListener.class);
 
         doAnswer((i) -> {
-            actionListenerArgumentCaptor.getValue().onFailure(new RuntimeException("Cannot upload to remote"));
+            // For async write action listener will be called from different thread, replicating same behaviour here.
+            new Thread(new Runnable() {
```

Comment:
wait for thread for finish either via thread.join or use latch otherwise, it can become flaky if it this thread has not executed first.

Reply:
Latch in `writeGlobalMetadata` will be released after the execution of this thread only, as in this thread we are going to call the `onFailure` of LatchedActionListener which will unblock the method execution. So assertion will happen after the execution of this thread only.

So test will not have flaky behavior. I ran it with 1000 iteration as well, and it is passing.